src/test/java/com/kodilla/ecommercee/domain/OrderRepositoryTestSuite.java

```diff
@@ -0,0 +1,195 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import com.kodilla.ecommercee.repository.UserRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@Transactional
+@SpringBootTest
+public class OrderRepositoryTestSuite {
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    private User user;
+    private User user2;
+    private User user3;
+    private Cart cart;
+    private Cart cart2;
+    private Cart cart3;
+    private Order order;
+    private Order order2;
+
+
+    private void createData() {
+        user = new User(0, "testUser", false, new UserKey(), new ArrayList<>(), new ArrayList<>());
+        cart = new Cart(0, new ArrayList<>(), user);
+        order = new Order(0, user, cart);
+        user.addCart(cart);
+        user.addOrder(order);
+    }
+
+    private void createAdditionalData() {
+        user2 = new User(0, "testUser nr2", false, new UserKey(), new ArrayList<>(), new ArrayList<>());
+        cart2 = new Cart(0, new ArrayList<>(), user2);
+        order2 = new Order(0, user2, cart2);
+        user2.addCart(cart2);
+        user2.addOrder(order2);
+    }
+
+    private void deleteData() {
+        try {
+            orderRepository.delete(order);
+            cartRepository.delete(cart);
+            userRepository.delete(user);
+        } catch (Exception e) {
+            System.out.println("Something went wrong! Error: " + e);
+        }
+    }
+
+    @Test
+    public void testOrderRepositorySave() {
+        //Given
+        createData();
+
+        //When
+        userRepository.save(user);
```

Comment:
co testujemy? zachowywanie encji user, cart i order? To ma byc test jednostkowy, wiec w sekcji When jest tylko save(order)